### PR TITLE
[Whisper] Add rescaling function with `do_normalize`

### LINKED ISSUE #19888: Rescale layer in whisper processor
### Feature request

Whisper processor does not currently rescale to the expected [-1, 1) that it requires.

### Motivation

Consistency between model processor layers.

### Your contribution

-

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Fixes #19888, by allowing the user to `normalise` the input audio before computing the MEL spectrogra,.

### CODE REVIEW

File: src/transformers/models/whisper/feature_extraction_whisper.py [hunk 1]
```diff
@@ -311,6 +338,12 @@ def __call__(
         # make sure list is in array format
         input_features = padded_inputs.get("input_features").transpose(2, 0, 1)
 
+        # zero-mean and unit-variance normalization
+        if do_normalize:
+            padded_inputs["input_features"] = self.zero_mean_unit_var_norm(
+                padded_inputs["input_features"], attention_mask=None, padding_value=self.padding_value
```
[the remark this answers is not in the document]
Comment (ArthurZucker) [reply src/transformers/models/whisper/feature_extraction_whisper.py:344]: At this point there are no attention mask. Feel free to pull from this once merged!

**Review by sgugger — COMMENTED** [review]
Thanks for your work on this!

File: src/transformers/models/whisper/tokenization_whisper.py [hunk 2]
```diff
@@ -479,8 +479,11 @@ def _convert_token_to_id(self, token):
         return self.encoder.get(token, self.encoder.get(self.unk_token))
 
     def _convert_id_to_token(self, index):
-        """Converts an index (integer) in a token (str) using the vocab."""
-        return self.decoder.get(index, self.decoder.get(self.unk_token_id))
+        """
+        Converts an index (integer) in a token (str) using the vocab. Whisper's base tokenizer always decodes OOV
+        tokens as "", thus we do not use the `unk_token` here.
+        """
+        return self.decoder.get(index, "")
```
Comment (sanchit-gandhi) [inline src/transformers/models/whisper/tokenization_whisper.py:486]: Don't believe this change is intended for this PR! We saw this fixed in https://github.com/huggingface/transformers/pull/21250. Might need to rebase

**Review by sanchit-gandhi — DISMISSED** [review]
Functionality LGTM, thanks for working on this! A few small things needed to fix this PR before merge!

**Review by sgugger — APPROVED** [review]
Thanks again!

**Comment by ArthurZucker** [comment]
The test works locally, merging 😉
